iwyu_preprocessor.cc

```diff
@@ -209,6 +217,19 @@ void IwyuPreprocessorInfo::HandlePragmaComment(SourceRange comment_range) {
     return;
   }
 
+  if (HasOpenBeginKeep(this_file_entry)) {
+    if (MatchOneToken(tokens, "end_keep", 1, begin_loc)) {
+      ERRSYM(this_file_entry) << "end_keep pragma seen\n";
+      SourceLocation temp = begin_keep_location_stack_.top();
```

Comment:
`temp` is a little vague. Maybe `begin_keep_loc`?

Also, don't we know that `GetFileEntry(temp) == this_file_entry` by virtue of `HasOpenBeginExports` being true? So you can use `this_file_entry` for `keep_location_ranges` key below.